### ISSUE TITLE
obtener imágenes con función REST[Mejora]

### ISSUE THREAD

**Issue by heber-solis** [FridaSupport/CognitiveTestingMX#182]
**Es una idea de mejora relacionada a un problema actual? por favor descríbe.**
Al hacer una petición REST me regresa una imagen, se puede obtener esa imagen con FRIDA? 

**Describe la idea o solución ideal**
guardar la imagen en alguna ruta

**Describe alternativas que hayas considerado**
Hay otras alternativas de solución para el problema?

**Additional context**
Actualmente me regresa la imagen así
![image](https://user-images.githubusercontent.com/54727278/149819686-d515c2c9-bcb2-492b-b5f7-e3618967b360.png)

esto es lo que regresa postman
![image](https://user-images.githubusercontent.com/54727278/149819757-bb960571-d2b9-4f4e-adfc-3c73fb5505db.png)

**Comment by AlanArteche** [comment]
Qué tal, @heber-solis , ¿podrías compartir el log de ejecución para validar si la imagen enviada por parte de la API está en Base 64 y es posible hacer la decodificación con la instrucción B64 (https://admin.fridaplatform.online/Documentation/Readers/MixReader/B64)?

**Comment by heber-solis** [comment]
[log-2022011716012642.txt](https://github.com/FridaSupport/CognitiveTestingMX/files/7890467/log-2022011716012642.txt)
hola que tal, si claro, igual deja voy intentando esa alternativa de una vez, muchas gracias!

**Comment by AlanArteche** [comment]
Revisando la documentación de la API que se está utilizando (remove.bg), se detecta que la API no responde con la imagen codificada en base 64, así que la instrucción para decodificarla no funciona en este caso.

![image](https://user-images.githubusercontent.com/48565879/150021055-4f392c17-e385-4788-ba7c-c7239ab95d45.png)

El formato de la respuesta por parte de la API es la imagen en formato binario o un Byte Stream.

Ya se logró recibir la imagen de la API y guardarla en alguna ruta local específica, pero se estará evaluando si es necesario aplicar modificaciones a la instrucción REST adicionalmente a la creación de una nueva instrucción que permita guardar una imagen dada en formato Byte Stream.

![image](https://user-images.githubusercontent.com/48565879/150021342-2d32816a-2d3d-44d2-a36a-4dcd9a181eff.png)

**Comment by heber-solis** [comment]
de acuerdo muchas gracias, entiendo que como está la instrucción no es posible obtener la imagen cierto?

**Comment by AlanArteche** [comment]
Correcto, al momento no sería posible. La propuesta es aplicar un cambio a la instrucción REST y la creación de una sintáxis adicional para la instrucción B64 que reciba el string de una imagen codificada en Base 64 (La instrucción REST en este caso estaría guardándola ya en este formato en la variable **response**) y se le indique un path local donde se guardará la imagen ya en su formato original.

**Comment by AlanArteche** [comment]
Qué tal, @heber-solis , buenas tardes, te comento que ya se implementaron los cambios. Se generó la sintáxis **B64 SaveImage** la cual de entrada toma un string en Base64 (Como el que ya guardaría correctamente la instrucción REST con la respuesta de la API que estás utilizando) y lo convierte a imagen en un path local. Los formatos de la imagen puede ser **BMP, EMF, GIF, ICON, JPEG, PNG, TIFF, WMF.** 

Ya se encuentra documentada la nueva sintáxis y ya se generó una nueva versión que la incluye, ¿requieres que te envíe update? Saludos.

**Comment by heber-solis** [comment]
hola, muchas gracias! Si, por favor

**Comment by AlanArteche** [comment]
Listo, ya se te asignó la nueva versión, favor de continuar con la prueba original. Quedamos a la espera de los resultados.

**Comment by heber-solis** [comment]
hola, disculpa me sale este error al intentar hacer la prueba
[log.txt](https://github.com/FridaSupport/CognitiveTestingMX/files/7916740/log.txt)

**Comment by AlanArteche** [comment]
Qué tal, @heber-solis , me suena a que pudiera ser un tema de la cultura de tu máquina, ¿puedes probar con el siguiente path y comentar si se presentó el mismo mensaje de error?
**C:\\Users\\MX02689\\Desktop\\MyImage.png**

**Comment by heber-solis** [comment]
así si me dejó, que es lo que se tendría que cambiar para que me funcione sin el problema?

**Comment by eva-ti** [comment]
@AlanArteche, @heber-solis, cuál es el estatus de este tema?

**Comment by AlanArteche** [comment]
Qué tal, @eva-ti , como se puede leer, el estatus es que la mejora fue implementada y probada, sólo quedó una duda sobre el formato de una ruta para evitar que cause problemas a futuro y no quede del lado del usuario cuidar este detalle pero no se ha revisado.
